src/app/shared/check-cpf.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive } from '@angular/core';
+import { NG_VALIDATORS, Validator, AbstractControl } from '@angular/forms';
+
+@Directive({
+  selector: '[appCheckCpf]',
+  providers: [{
+    provide: NG_VALIDATORS,
+    useExisting: CheckCpfDirective,
+    multi: true
+  }],
+})
+export class CheckCpfDirective implements Validator {
+  validate(control: AbstractControl): { [key: string]: any } | null {
+    if (!control.value) return null;
+    const checkDuplicateDigit = digits => digits.split('').every((val, i, arr) => val === arr[0]);
+
+    const calculateCpfNumbers = (total) => (result, num, i) => result + (num * total--);
+
+    const checkDigit = (numbers, total) => numbers.split('').reduce(calculateCpfNumbers(total), 0);
+
+    const getDigit = (num) => (num > 1) ? (11 - num) : 0;
+
+    const isEqual = (a) => (b) => b === a
+
+    const isValid = (cpf) => {
+      const numbers = cpf.substring(0, 9);
```

Comment:
Always **type** the vars on angular, its a good practice ;)

```suggestion
      const numbers: number = cpf.substring(0, 9)
```